### ISSUE TITLE
Possibly missing role roles/artifactregistry.admin when trying to create IAM on artifact registry

### ISSUE THREAD

**Issue by gnthibault** [terraform-google-modules/terraform-google-project-factory#945]
### TL;DR

Dear all,
Using this module, I managed to properly deploy a new project on GCP, as well as some additional ressources, such as an artifact registry, to hold docker containers using this module: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository

The AR is properly deployed, however, everytime I try to set some specific roles on this registry, I get the following error:
│ Error: Error retrieving IAM policy for artifactregistry repository "projects/my-project/locations/europe-west1/repositories/my-repo": googleapi: Error 403: The caller does not have permission
│ 
│   with module.my_app.google_artifact_registry_repository_iam_member.member_runner,
│   on ../../modules/my_app/container_registry.tf line 28, in resource "google_artifact_registry_repository_iam_member" "member_runner":
│   28: resource "google_artifact_registry_repository_iam_member" "member_runner" {
│ 
╵
╷
│ Error: Error retrieving IAM policy for artifactregistry repository "projects/my-project/locations/europe-west1/repositories/my-repo": googleapi: Error 403: The caller does not have permission
│ 
│   with module.my_app.google_artifact_registry_repository_iam_member.member_basic_admin,
│   on ../../modules/my_app/container_registry.tf line 36, in resource "google_artifact_registry_repository_iam_member" "member_basic_admin":
│   36: resource "google_artifact_registry_repository_iam_member" "member_basic_admin" {
│ 
╵
╷
│ Error: Error retrieving IAM policy for artifactregistry repository "projects/my-project/locations/europe-west1/repositories/my-repo": googleapi: Error 403: The caller does not have permission
│ 
│   with module.my_app.google_artifact_registry_repository_iam_member.member_builder,
│   on ../../modules/my_app/container_registry.tf line 44, in resource "google_artifact_registry_repository_iam_member" "member_builder":
│   44: resource "google_artifact_registry_repository_iam_member" "member_builder" {

The access right are set using this TF module:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository_iam#google_artifact_registry_repository_iam_member

and looks like this:
```
resource "google_artifact_registry_repository_iam_member" "member_runner" {
  project       = google_artifact_registry_repository.my-repo.project
  location      = google_artifact_registry_repository.my-repo.location
  repository    = google_artifact_registry_repository.my-repo.project
  role          = "roles/artifactregistry.reader"
  member        = google_service_account.deploy_service_account.member
}

```

### Expected behavior

I would expect to have the tf service account granted with the proper role, most likely roles/artifactregistry.admin

### Observed behavior

As of today, I see in my configuration that the TF SA has:
Owner
Security Admin
Service Account Admin
Compute Network Admin
Logs Configuration writer

### Terraform Configuration

```hcl
Not sure how I can provide a self contained configuration, will work on that
```

### Terraform Version

```sh
OpenTofu v1.8.1
```

**Comment by gnthibault** [comment]
The issue was simply related to wrong parameter passing in the ressource
repository    = google_artifact_registry_repository.my-repo.project
Should have been
repository    = google_artifact_registry_repository.my-repo.name
